proctor-common/src/main/java/com/indeed/proctor/common/ForceGroupsOptionsStrings.java

```diff
@@ -62,26 +71,105 @@ public static ForceGroupsOptions parseForceGroupsString(@Nullable final String f
             }
             //  bucketValueStart should now be the index of the minus sign or the first digit in a run of digits going to the end of the word
             final String testName = piece.substring(0, bucketValueStart).trim();
-            final String bucketValueStr = piece.substring(bucketValueStart);
+            final String bucketValueStr = piece.substring(bucketValueStart).trim();
             try {
                 final Integer bucketValue = Integer.valueOf(bucketValueStr);
                 builder.putForceGroup(testName, bucketValue);
+                if (bucketAndPayloadValuesStr.length == FORCE_PARAMETER_MAX_SIZE) {
+                    final Payload payloadValue = parseForcePayloadString(bucketAndPayloadValuesStr[FORCE_PARAMETER_PAYLOAD_IDX]);
+                    if (payloadValue != null){
+                        builder.putForcePayload(testName, payloadValue);
+                    }
+                }
             } catch (final NumberFormatException e) {
-                LOGGER.error("Unable to parse bucket value " + bucketValueStr + " as integer", e);
+                LOGGER.error("Unable to parse bucket value " + bucketAndPayloadValuesStr[FORCE_PARAMETER_BUCKET_IDX] + " as integer", e);
```

Comment:
Why do you replace `bucketValueStr` with a different value here? It seems we still parse `bucketValueStr` as integer at line 76.

Reply:
Reverted back to bucketValueStr - forgot to switch this back after some iterations of changes.